src/number_theory/modular_forms/slash_invariant_forms.lean

```diff
@@ -0,0 +1,189 @@
+begin
+  rw ←modular_form.slash_action_eq'_iff,
+  simp,
+end
+
+instance [slash_invariant_form_class F Γ k] : has_coe_t F (slash_invariant_form Γ k) :=
+⟨λ f, { to_fun := f, slash_action_eq' := slash_action_eqn f }⟩
+
+@[simp] lemma slash_invariant_form_class.coe_coe [slash_invariant_form_class F Γ k] (f : F) :
+  ((f : slash_invariant_form Γ k) : ℍ → ℂ) = f := rfl
+
+instance has_add : has_add (slash_invariant_form Γ k) :=
+⟨ λ f g,
+  { to_fun := f + g,
+    slash_action_eq' := λ γ, by convert slash_action.add_action k γ (f : ℍ → ℂ) g; simp } ⟩
+
+@[simp] lemma coe_add (f g : slash_invariant_form Γ k) : ⇑(f + g) = f + g := rfl
+@[simp] lemma add_apply (f g : slash_invariant_form Γ k) (z : ℍ) : (f + g) z = f z + g z := rfl
+
+instance has_zero : has_zero (slash_invariant_form Γ k) :=
+⟨ { to_fun := 0,
+    slash_action_eq' := slash_action.mul_zero _} ⟩
+
+lemma nsmul_coe [slash_invariant_form_class F Γ k] (f : F) (c : ℕ) :
+  c • (f : ℍ → ℂ) = (c : ℂ) • f :=
+begin
+ simp only [nsmul_eq_mul],
+ congr,
+end
+
+lemma zsmul_coe [slash_invariant_form_class F Γ k] (f : F) (c : ℤ) :
+  c • (f : ℍ → ℂ) = (c : ℂ) • f :=
+begin
+ simp only [zsmul_eq_mul],
+ congr,
+end
+
+instance has_csmul : has_smul ℂ (slash_invariant_form Γ k) :=
+⟨ λ c f, {to_fun := c • f,
+    slash_action_eq' := by {intro γ, convert slash_action.smul_action k γ ⇑f c,
+    exact (f.slash_action_eq' γ).symm}}⟩
+
+@[simp] lemma coe_csmul (f : slash_invariant_form Γ k) (n : ℂ) : ⇑(n • f) = n • f := rfl
+@[simp] lemma csmul_apply (f : slash_invariant_form Γ k) (n : ℂ) (z : ℍ) :
+  (n • f) z = n • (f z) := rfl
+
+instance has_nsmul : has_smul ℕ (slash_invariant_form Γ k) :=
+⟨ λ c f, ((c : ℂ) • f).copy (c • f) (nsmul_eq_smul_cast _ _ _)⟩
+
+@[simp] lemma nsmul_apply (f : slash_invariant_form Γ k) (n : ℕ) (z : ℍ) :
+   (n • f) z = n • (f z) := rfl
+
+instance has_zsmul : has_smul ℤ (slash_invariant_form Γ k) :=
+⟨ λ c f, ((c : ℂ) • f).copy (c • f) (zsmul_eq_smul_cast _ _ _)⟩
+
+@[simp] lemma zsmul_apply (f : slash_invariant_form Γ k) (n : ℤ) (z : ℍ) :
+   (n • f) z = n • (f z) := rfl
+
+instance has_neg : has_neg (slash_invariant_form Γ k) :=
+⟨ λ f,
+  { to_fun := -f,
+    slash_action_eq' := λ γ, by simpa [modular_form.subgroup_slash,
+      modular_form.neg_slash] using f.slash_action_eq' γ } ⟩
+
+@[simp] lemma coe_neg (f : slash_invariant_form Γ k) : ⇑(-f) = -f := rfl
+@[simp] lemma neg_apply (f : slash_invariant_form Γ k) (z : ℍ) : (-f) z = - (f z) := rfl
+
+instance has_sub : has_sub (slash_invariant_form Γ k) := ⟨ λ f g, f + -g ⟩
+
+@[simp] lemma coe_sub (f g : slash_invariant_form Γ k) : ⇑(f - g) = f - g := rfl
+@[simp] lemma sub_apply (f g : slash_invariant_form Γ k) (z : ℍ) : (f - g) z = f z - g z := rfl
+
+instance : add_comm_group (slash_invariant_form Γ k) :=
+fun_like.coe_injective.add_comm_group _ rfl coe_add coe_neg coe_sub
+  (λ f n, nsmul_eq_smul_cast _ n f) (λ f n, zsmul_eq_smul_cast _ n f)
+
+lemma coe_zero : ⇑(0 : slash_invariant_form Γ k) = (0 : ℍ → ℂ) := rfl
```

Comment:
For organizational purposes, can you please put this right after `has_zero`, just as you have put the other `coe` lemmas right after the appropriate instance?

Reply:
And can you please make it a simp-lemma?